### ISSUE TITLE
IOS Implementation

### ISSUE THREAD

**Issue by Psycotype** [ethan-ou/offline_speech_recognition#1]
Hello, I am new to ASR and speech recognition tools and I was wondering if development for this package is still ongoing. Also, is it possible for me to try to bind IOS myself using this package as a basis? If so, can you point me to a general idea as to how to do so? Thank you!

**Comment by ethan-ou** [comment]
Yes I was hoping to get back to updating this project some time in the future. The end goal is to merge it back to the Vosk repository where it can be maintained.

To start off, have a look at the [Flutter docs](https://flutter.dev/docs/development/packages-and-plugins/developing-packages#step-2c-add-ios-platform-code-swifthm) for how to start building the iOS section. You'll of course need XCode to build and flutter to run and test the plugin. If you're new to Swift I recommend just making a new Swift Flutter plugin from the plugin scaffold to see how the project is set up.

From there I'd go and look at integrating the existing Vosk library for iOS which is already written. The only thing you'll need to do is connect the API of that library to Flutter's message bridge. This is how Flutter will communicate with the platform native code.

Two resources I'd go to is the [plugins repo in the flutter org](https://github.com/flutter/plugins) and the [Vosk API demo project for iOS](https://github.com/alphacep/vosk-api/tree/master/ios). The plugins repo will show you how to connect the native code to the message bridge and the demo project is where you'll get the details of the API for the Vosk plugin. 

If you want me to dive deeper with you let me know.

**Comment by Psycotype** [comment]
Thanks for the information! I'm still quite new to Swift, so I'll work on getting a bit more comfortable with the language with your recommendation. After that, I'll definitely look further into the resources above, and let you know when I've made some progress!

**Comment by ethan-ou** [comment]
No worries. Hope you have fun learning!

**Comment by johngray1965** [comment]
Hello, I'll take a crack at the iOS wrapper.

**Comment by PruthviSooni** [comment]
> ive deeper with you

Hello @ethan-ou, Are you going to implement IOS part?
